pyiceberg/table/name_mapping.py

```diff
@@ -232,7 +234,9 @@ def mapping(self, nm: NameMapping, field_results: List[MappedField]) -> List[Map
 
     def fields(self, struct: List[MappedField], field_results: List[MappedField]) -> List[MappedField]:
         reassignments: Dict[str, int] = {
-            update.name: update.field_id for f in field_results if (update := self._updates.get(f.field_id))
+            update.name: update.field_id
+            for f in field_results
+            if f.field_id is not None and (update := self._updates.get(f.field_id))
```

Comment:
what is the rationale behind this change? Should we look at all the other places where `.field_id` is used?

Reply:
I see, is this change also due to the type checker?

Reply:
thanks for the explanation! cc @Fokko / @sungwy for another set of eyes